File: 071_Simplify_Path.java
// Given an absolute path for a file (Unix-style), simplify it.

// For example,
// path = "/home/", => "/home"
// path = "/a/./b/../../c/", => "/c"
// click to show corner cases.

public class Solution {
    public String simplifyPath(String path) {
        String[] paths = path.split("/");
        Deque<String> dirList = new ArrayDeque();
        for(String dirName: paths){
            if (dirName.equals("") || dirName.equals(".")){
                continue;
            }else if(dirName.equals("..") && dirList.size() != 0){
                dirList.removeLast();
            }else{
                dirList.add(dirName);
            }
        }
        if (dirList.size() == 0)    return "/";
        StringBuilder fianlPath = new StringBuilder();
        for (String dirName: dirList){
            fianlPath.append("/" + dirName);
        }
        return fianlPath.toString();
    }
}

File: python/071_Simplify_Path.py
# Given an absolute path for a file (Unix-style), simplify it.

# For example,
# path = "/home/", => "/home"
# path = "/a/./b/../../c/", => "/c"
# click to show corner cases.

# Subscribe to see which companies asked this question

# Runtime: 84 ms

class Solution(object):
    def simplifyPath(self, path):
        """
        :type path: str
        :rtype: str
        """
        res = []
        string = ""
        for ch in path:
            if ch == "/":
                if string == "..":
                    res and res.pop(-1)
                elif string != "." and string:
                    res.append(string)
                string = ""
            else:
                string+=ch
        if ch != "/":
            if string == "..":
                res and res.pop(-1)
            elif string != "." and string:
                res.append(string)
        res = "/".join(res)
        return "/"+res
if __name__ == '__main__':
    sol = Solution()
    print sol.simplifyPath("/..")


